packages/swingset-xsnap-supervisor/lib/supervisor-subprocess-xsnap.js

```diff
@@ -1,22 +1,26 @@
+// This file is only used to generate the published bundle, so
+// @endo/init is in devDependencies (and this package should have no
+// direct dependencies), but eslint doesn't know that, so disable the
+// complaint.
+
+/* eslint-disable import/no-extraneous-dependencies */
+
 /* global globalThis WeakRef FinalizationRegistry */
 import { assert, Fail } from '@agoric/assert';
 import { importBundle } from '@endo/import-bundle';
 import { makeMarshal } from '@endo/marshal';
 import { makeLiveSlots } from '@agoric/swingset-liveslots';
-import '../../types-ambient.js';
+// import '../../types-ambient.js';
 // grumble... waitUntilQuiescent is exported and closes over ambient authority
-import { waitUntilQuiescent } from '../../lib-nodejs/waitUntilQuiescent.js';
-import { makeGcAndFinalize } from '../../lib-nodejs/gc-and-finalize.js';
-import {
-  insistVatDeliveryObject,
-  insistVatSyscallResult,
-} from '../../lib/message.js';
+import { waitUntilQuiescent } from './waitUntilQuiescent.js';
+import { makeGcAndFinalize } from './gc-and-finalize.js';
```

Comment:
I think these should probably move to a shared package.

Reply:
Hrm, do you think it's worth it? I'm feeling bad about making, what, 5 new packages as part of #6596, especially if the plan requires them to all go into new repositories too. Also I need to make sure that the new package doesn't grow to include more dependencies (which is why I don't feel confident putting them in `@agoric/internal`, our current "stuff" package).

I suppose if I do create one, it should be called "platform GC tools" in some form.

Reply:
I think `@agoric/internal` is fine as long as we're using a deep import.

At the very least `waitUntilQuiescent` is deep linked from quite a few places already, and duplicated in the tests of swingset-liveslots. I am just concerned about copy paste code that later gets out of sync for some reason.

Reply:
Oh, you know, `waitUntilQuiescent` is sufficiently platform-specific that I don't think we actually *want* to share a version between Node.js -based users and XS -based ones (like this package). We have several extra `setImmediate` calls in `gc-and-finalize.js` that probably aren't necessary on XS, and there are comments about required Node.js versions (>11.x) in `waitUntilQuiescent.js` that aren't relevant to XS.

If we did move it to a separate shared package, but that package is also XS-specific, then this would be the only client, which would defeat the purpose.

I guess this implies that we might want to split out a `worker-local` package too, some day, and it might wind up owning the Node.js -flavored copy of these two utilities. I do see one user in cosmic-swingset. Maybe that's the flavor that could go into `@agoric/internal`.

I think I'm going to leave this copy alone.. trying to find a good home for it that's also somehow platform-specific sounds like too much work at this point.

Reply:
Oh I actually had not read the comment of waitUntilQuiescent, just the implementation, which is the correct one on any platform at this point. I honestly would not consider node 10 to still be relevant.

Reply:
The extra setImmediate in gc-and-finalize are surprising, and I believe are artifacts of either erroneous tests, or of engine bugs I'm unaware of. However they do not hurt at this point, on any platform.

Reply:
I am really not convinced we should have platform specific versions of GC tools. I know GC is implementation defined, but realistically it works sufficiently similarly in all implementations to be able to share helpers around it.